src/components/PeopleAndRoles/PeopleAndRoles.vue

```diff
@@ -1,102 +1,119 @@
 <template>
-  <v-card flat id="people-and-roles">
-    <!-- Header -->
-    <div class="header-container">
-      <v-icon color="#38598A">mdi-account-multiple-plus</v-icon>
-      <label class="font-weight-bold pl-2">People and Roles</label>
-    </div>
-
-    <!-- Instructional Text -->
-    <div class="role-container pt-6 px-4">
-      This application must include the following:
-      <ul>
-        <li>
-          <v-icon v-if="cpValid" color="blue" class="cp-valid">mdi-check</v-icon>
-          <v-icon v-else color="red" class="cp-invalid">mdi-close</v-icon>
-          <span class="ml-2">The Completing Party</span>
-        </li>
-        <li>
-          <v-icon v-if="incorpValid" color="blue" class="incorp-valid">mdi-check</v-icon>
-          <v-icon v-else color="red" class="incorp-invalid">mdi-close</v-icon>
-          <span class="ml-2">At least one Incorporator</span>
-        </li>
-        <li>
-          <v-icon v-if="dirValid" color="blue" class="dir-valid">mdi-check</v-icon>
-          <v-icon v-else color="red" class="dir-invalid">mdi-close</v-icon>
-          <span class="ml-2">At least one Director</span>
-        </li>
-      </ul>
-    </div>
-
-    <!-- Add Buttons -->
-    <div class="btn-container py-6 px-4">
-      <v-btn
-        id="btn-add-person"
-        outlined
-        color="primary"
-        :disabled="renderOrgPersonForm"
-        @click="initAdd([], IncorporatorTypes.PERSON)"
-      >
-        <v-icon>mdi-account-plus</v-icon>
-        <span>Add a Person</span>
-      </v-btn>
-      <v-btn
-        id="btn-add-corp"
-        outlined
-        color="primary"
-        class="ml-2"
-        :disabled="renderOrgPersonForm"
-        @click="initAdd([{ roleType: Roles.INCORPORATOR }], IncorporatorTypes.CORPORATION)"
-      >
-        <v-icon>mdi-domain-plus</v-icon>
-        <span>Add a Corporation or Firm</span>
-      </v-btn>
-      <v-btn
-        v-if="!cpValid"
-        id="btn-add-cp"
-        outlined
-        color="primary"
-        class="ml-2"
-        :disabled="renderOrgPersonForm"
-        @click="initAdd([{ roleType: Roles.COMPLETING_PARTY }], IncorporatorTypes.PERSON)"
-      >
-        <v-icon>mdi-account-plus-outline</v-icon>
-        <span>Add the Completing Party</span>
-      </v-btn>
-    </div>
-
-    <div class="list-container px-4">
-      <list-people-and-roles
-        :peopleAndRoles="getPeopleAndRoles"
-        :renderOrgPersonForm="renderOrgPersonForm"
-        :currentOrgPerson="currentOrgPerson"
-        :activeIndex="activeIndex"
-        :nextId="nextId"
-        :currentCompletingParty="currentCompletingParty"
-        @initEdit="initEdit($event)"
-        @addEdit="addEdit($event)"
-        @remove="remove($event)"
-        @undo="undo($event)"
-        @reset="reset()"
-        @removeCpRole="removeCpRole()"
-      />
-    </div>
-  </v-card>
+  <div id="people-and-roles">
+
+    <confirm-dialog
```

Comment:
Yes, this is a second instance of this dialog, but it has a different message, is used differently, and it was really messy (nigh impossible) to have a single instance called from a component and its grandparent/grandchild. So here we are... easier and cleaner.

Reply:
i understand!